2.3/configuration/the-basics.md

```diff
@@ -0,0 +1,183 @@
+# The Basics
+
+[[toc]]
+
+## Creating configuration
+
+Your ls-lint configuration should be located in a `.ls-lint.yml` file in your project **root** directory and provides two options:
+
+<div style="color:#A2A2A2; font-size:12px;">
+    .ls-lint.yml
+</div>
+
+```yaml
+ls: 
+    ... 
+
+ignore: 
+    ...
+```
+
+`ls` defines the structure of your project directories with all their rules for your extensions, sub-extensions and directories while `ignore` can ignore some of these files and directories completely
+
+:::tip Pro Tip
+You can change the location of your `.ls-lint.yml` file and your work directory with the cli options: [Options](/2.3/configuration/the-command-line-interface.md#options.md)
+:::
+
+## Extensions & sub-extensions
+
+You might ask yourself - what the hell are sub-extensions? ls-lint describes file extensions like `.js` or `.html` as extensions while `.d.ts`, `.umd.js` or `.local.build.js` file extensions are described as sub-extensions.
+
+:::tip Wildcard extensions
+You can also use wildcard extensions to simplify your configuration: [Wildcard extensions](/2.3/configuration/the-basics.md#linting-wildcard-extensions)
+:::
+
+## Global configuration
+
+A global configuration for your project is optional but could be the best solution. Imagine you want to define that **all** of your `.js`, `.ts` and `.d.ts` project files must be named in the `kebab-case` style but you want to ignore your `.git` and `node_modules` directories - this can be solved like this:
+
+<div style="color:#A2A2A2; font-size:12px;">
+    .ls-lint.yml
+</div>
+
+```yaml
+ls:
+    .js: kebab-case
+    .ts: kebab-case
+    .d.ts: kebab-case
+
+ignore: 
+    - .git 
+    - node_modules
+```
+
+Pretty easy, isn`t it? 
+
+:::tip Rules Rules Rules
+Don`t worry, there are many more rules to use: [All rules](/2.2/configuration/the-rules.md)
+:::
+
+## Linting directory names
+
+You also can define rules for directories by the `.dir` definition
+
+```yaml
+ls:
+  packages/src:
+    .dir: kebab-case # applies for the current directory and all their subdirectories
+    .js: kebab-case
+```
+
+## Linting wildcard extensions
+
+You can use wildcard extensions for files in your **ls** configuration. With this support, you are no longer forced to add any possible file extension to your .ls-lint.yml configuration file, for instance:
+
+```yaml
+ls:
+  .*: snake_case
+  .js: kebab-case
+  .*.js: kebab-case
```

Comment:
Here the example is not easy to understand

As you are applying kebab for both .js

You should add that this allow you to use

Pascal for test.js, but kebab for all other .js

And something like without it, it would have been impossible to use .test.js sub-extension

Reply:
```yaml
ls:
  .*: snake_case # match all extensions but .js and .*.js files
  .js: kebab-case # match all .js files
  .*.js: kebab-case # match all .*.js files (e.g. .test.js, .service.js)
```

Reply:
It's clearer, yes